homeassistant/components/google_generative_ai_conversation/config_flow.py

```diff
@@ -126,53 +119,84 @@ async def async_step_init(
     ) -> ConfigFlowResult:
         """Manage the options."""
         if user_input is not None:
-            return self.async_create_entry(
-                title="Google Generative AI Conversation", data=user_input
-            )
-        schema = google_generative_ai_config_option_schema(self.config_entry.options)
+            return self.async_create_entry(title="", data=user_input)
+        schema = await google_generative_ai_config_option_schema(
+            self.hass, self.config_entry.options
+        )
         return self.async_show_form(
             step_id="init",
             data_schema=vol.Schema(schema),
         )
 
 
-def google_generative_ai_config_option_schema(
+async def google_generative_ai_config_option_schema(
+    hass: HomeAssistant,
     options: MappingProxyType[str, Any],
 ) -> dict:
     """Return a schema for Google Generative AI completion options."""
-    if not options:
-        options = DEFAULT_OPTIONS
+    api_models = await hass.async_add_executor_job(partial(genai.list_models))
+
+    models: list[SelectOptionDict] = [
+        SelectOptionDict(
+            label="Gemini 1.0 Pro (free in US)",
+            value="models/gemini-pro",
+        ),
+        SelectOptionDict(
+            label="Gemini 1.5 Flash (fastest)",
+            value="models/gemini-1.5-flash-latest",
+        ),
+    ]
+    models.extend(
+        SelectOptionDict(
+            label=api_model.display_name,
+            value=api_model.name,
+        )
+        for api_model in sorted(api_models, key=lambda x: x.display_name)
+        if (
+            api_model.name
+            not in (
+                "models/gemini-pro",
+                "models/gemini-1.0-pro",  # duplicate
+                "models/gemini-1.5-flash-latest",
+            )
+            and "generateContent" in api_model.supported_generation_methods
```

Comment:
Thanks for including this. We need to figure out how to exclude the vision models too. Those won't work with text only.